src/mgm.rs

```diff
@@ -86,44 +90,167 @@ pub enum MgmType {
     Protected = 2,
 }
 
+/// The algorithm used for the MGM key.
+pub trait MgmKeyAlgorithm:
+    BlockCipher + BlockDecrypt + BlockEncrypt + KeyInit + private::Seal
+{
+    /// The KeySized used for this algorithm
+    const KEY_SIZE: u8;
```

Comment:
@tony-iqlusion Is it possible to get this out of `cipher`'s generics? I didn't have a huge amount of success there, as it seems to be that yubikey.rs needs this in a const context, but the generics are not const. I understand why key size for rust crypto traits might not be possible to make const, but I want to know if I didn't miss anything.

Reply:
`KeyInit` has a supertrait bound on `KeySizeUser`: https://docs.rs/crypto-common/latest/crypto_common/trait.KeySizeUser.html

That has an associated [`typenum::Unsigned`] you can get the `USIZE` constant for, something like:

```rust
<Cipher as KeySizeUser>::KeySize::USIZE
```

It might be possible to get away with const generics for all that in a const context if you're only accessing the associated constant, but it might need an MSRV bump. I believe they expanded generic support for `const fn` recently. Or I could be wrong.

Reply:
which would you prefer a version bump or the hard-codes
(I am very agnostic here)?

Reply:
We are forced to do an MSRV bump due to upstream dependencies moving to `hybrid-array`; this was done in #582. So we will end up doing a version bump in the next release.